cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -63,6 +68,7 @@
   private final Map<String, SparkSpecification> sparks = Maps.newHashMap();
   private final Map<String, WorkflowSpecification> workflows = Maps.newHashMap();
   private final Map<String, ServiceSpecification> services = Maps.newHashMap();
+  private final Map<String, ScheduleConfigurer> schedules = Maps.newHashMap();
```

Comment:
Same schedule can be configured with multiple programs. ScheduleSpecification maintains the map of Schedules and associated set of programs. We want to delay the creation of the specification till all the programs are added. Hence configured is used here which keeps adding the programs to the schedules and at the end we create specifications from it. Please let me know in case any questions.